src/client/src/rt-theme/ThemeState.tsx

```diff
@@ -0,0 +1,96 @@
+import React from 'react'
+import { ThemeProvider } from './ThemeProvider'
+import { Theme, themes } from './themes'
+
+type ThemeName = 'light' | 'dark'
+
+interface ThemeSelector {
+  name?: ThemeName
+  /**
+   * An unused property — this parallel approach would support
+   * custom branding on child components — assuming they
+   * subscribe to the struture of our Theme type
+   *
+   * @evanrs (8/14/18)
+   */
+  theme?: Theme
+}
+
+interface ThemeStateProps extends ThemeSelector {
+  onChange?: (value: ThemeSelector) => void
+}
+
+export interface ThemeStateValue extends ThemeSelector {
+  setTheme: (options: ThemeSelector) => void
+}
+
+const { Provider: ContextProvider, Consumer: ContextConsumer } = React.createContext(null)
+
+/**
+ * Set default theme and allow descendants to update selected theme.
+ *
+ * @example
+ *  <ThemeState name={light | dark} onChange={this.handleThemeChange} />
+ */
+class ThemeStateProvider extends React.Component<ThemeStateProps> {
+  render() {
+    return <ContextConsumer children={this.renderThemeStateManager} />
+  }
+
+  renderThemeStateManager = (context: ThemeStateValue | null) => {
+    return <ThemeStateManager context={context} {...this.props} />
+  }
+}
+
+class ThemeStateManager extends React.Component<ThemeStateProps & { context: ThemeStateValue }, ThemeStateValue> {
```

Comment:
We should avoid naming things with words like "manager". Its a little wooly.

Reply:
Yup, make your suggestion.